Automattic/jetpack
# Social | Improve reconnection flow for broken connections

Improve reconnection flow for broken connections by automatically opening the connection window

This changeset turned out to be bigger than anticipated due to these reasons:
- Now that connection/reconnection flow can be fired from within the list and from the modal, the keyring result needs to be shared between the list and the modal. Thus, I added the keyring result to the store to make it available both on the admin page and in the editor.
- Sometimes, users may close the modal while the reconnection/disconnection is in progress. In that case, the request may complete in the background and thus setting the keyring result. So, we need to ensure that the keyring result is cleaned up on modal close and also fire up the modal if the reconnection is fired from inside the modal and then the modal gets closed.
- We no longer need the `defaultExpandedService` logic when opening the modal for reconnection. Thus, that code has been cleaned up.

## Proposed changes:

* Make the reconnect button work like the Disconnect and Connect buttons together

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Ensure that feature flag is ON - `define( 'JETPACK_SOCIAL_USE_ADMIN_UI_V1', true );`
- Test the following for both Jetpack and Social plugin
    - Goto settings page
    - Confirm that everything works as before
    - Now to simulate a broken connection on page log,  add `'status' => 'broken'` on line 265 in `projects/packages/publicize/src/class-publicize.php`
    - Goto settings page again
    - Now you should see the broken connection notice
    - Click on "Reconnect"
    - Confirm that the connection is deleted and the window for a new connection for that specific social network is fired.
    - Confirm that it works the same for Mastodon, automatically picking up the existing username
    - Open the modal for broken connections
    - Confirm that you see the broken connection notice below the account name
    - Click on "Reconnect", confirm the same behavior as above reconnection
    - Close the modals unexpectedly to confirm it works as expected.



https://github.com/Automattic/jetpack/assets/18226415/b2179da4-50a5-436f-a20e-0fdf385a28f6
